balancer/rls/balancer.go

```diff
@@ -263,17 +266,16 @@ func (b *rlsBalancer) UpdateClientConnState(ccs balancer.ClientConnState) error
 	// channels, we also swap out the throttling state.
 	b.handleControlChannelUpdate(newCfg)
 
-	// If the new config changes the size of the data cache, we might have to
-	// evict entries to get the cache size down to the newly specified size.
-	if newCfg.cacheSizeBytes != b.lbCfg.cacheSizeBytes {
-		b.dataCache.resize(newCfg.cacheSizeBytes)
-	}
-
 	// Any changes to child policy name or configuration needs to be handled by
 	// either creating new child policies or pushing updates to existing ones.
 	b.resolverState = ccs.ResolverState
 	b.handleChildPolicyConfigUpdate(newCfg, &ccs)
 
+	resizeCache := false
+	if newCfg.cacheSizeBytes != b.lbCfg.cacheSizeBytes {
+		resizeCache = true
+	}
```

Comment:
```go
// Resize the cache if the size in the config has changed.
resizeCache := newCfg.cacheSizeBytes != b.lbCfg.cacheSizeBytes
```

(Comment optional, but explains what could be seen as "more confusing logic", and still uses fewer lines.)

Reply:
Done. Thanks.